### ISSUE TITLE
How to use the position editor?

### ISSUE THREAD

**Issue by Menghus** [JulianOzelRose/TRR-SaveMaster#9]
Hello again! I first want to thank you for your great work with this mod and I appreciate your effort :)

I've seen the position editor in the new update and all I see is coordinates that you can change. How do I know where Lara will end up in the map exactly if I edit the coordinates? Sorry about my ignorance, I hope you can give me a simple explanation of how to do it.

I also asked you earlier about the possibility of adding an option to change enemy health and weapon damage. Can you tell me about the current progress with that and if it's possible to add it, about when can we expect an update? Sorry, I really don't mean to pressure you but I'm just curious and really looking forward to this feature (assuming it's even possible of course). Thanks!

**Comment by JulianOzelRose** [comment]
Hello, it's no problem I enjoy doing this stuff :)

About the position editor... Basically the X and Z coordinates work exactly how you would expect them to; decreasing X moves Lara to the left, increasing it moves her to the right. Increasing Z moves her forward, decreasing it moves her backwards. The Y-coordinate is a bit counter-intuitive; increasing it lowers her height, and decreasing it increases her height. All of these coordinates are in multiples of 256, which is about 1 game "unit". So be sure to either use the up/down arrows in the number box, or if you manually input coordinates, make sure that they are multiples of 256.

For the "Direction" value... I'm actually not entirely sure what it is or how it is calculated. My best guess is it might have something to do with view angles. All I know is that if you teleport Lara large distances and don't change that variable, she will clip off the map. But if you have the coordinate set of point A and B, you can teleport her back and forth without having to calculate it. If you are teleporting her short distances, like to get across a bridge or something like that, you typically don't have to worry about that variable. Hope that helps.

For the other things you're asking about... the ability to change global variables like weapon damage is next on my to do list. Since I only just finished the position editor feature, I've only just begun to look into it. I do believe it is possible to change the global variables. If they are stored in the `savegame.dat` file, I should be able to figure it out pretty quickly.

For changing enemy health, I think this sounds very interesting. I found out how to change enemy health in TR3 classic a while ago, so I know it is definitely possible. I think it would be relatively easy to get it working for final bosses. It would be a lot trickier for other "regular" entities that just spawn and die, like raptors for example. It might be especially difficult for entities that just respawn, like the wasps in the Lost City of Tinnos level. Could you elaborate a bit on what would make this feature useful to you? Are you thinking of just editing final bosses' health, or all entities?

**Comment by Menghus** [comment]
I see, thank you for the explanation of how to use the position editor.

I've seen a list of how much damage each weapon does per shot and how much HP each enemy has and I've noticed that in many combat encounters (especially in TR2 & TR3), there is an imbalance between how much HP an entity has and how much damage a weapon you choose does. Here's an example: In TR2, a tiger has 20 HP. If I want to use a shotgun which does 18 HP of damage per shot if the shot hits perfectly without any miss, I have to use 2 shells just for that 2 HP shortage. This is something that the original devs did not seem to think about when deciding the value of these things (HP and weapon damage).

That was just one example, but there are many cases with most weapon and enemies where this imbalance is present and the idea I had in my mind was to edit the HP and weapon damage for many entities and weapons to make it balanced as much as possible to avoid unnecessary damage loss.

I don't know if I misunderstood your comment, but it seems like you're implying that adding the option to edit weapon damage is more simple and less complicated than adding the ability to change the health of all entities and not just bosses. If that's the case then for me personally, the possibility of changing weapon damage is more important to me than changing enemy HP so even if only the weapon damage was added in an upcoming patch, that would be enough for me and I will be totally fine, it's still a significant improvement compared to the original state of the game. If it would still be possible to also add the option to change the health for all entities without too much hassle to make it work, that would of course be a big plus and very much appreciated but as I said, not as necessary.

Once that gets completed (again, assuming it will work of course), I would like to ask you if you also can make it possible to add an option to make the monks take damage from the enemies in Barkhang Monastery that carries the Uzis and not just from the guys that carries handguns? I don't know if you are aware of this, but the monks can only be killed by the latter guy while they are invincible against the Uzi guys. It's worth mentioning that this feature was added in the New Game+ in TR2 by the devs behind the remaster but not in normal mode, so you might just need to take that feature from the NG+ and add it to normal mode, nothing else.

Appreciate your help!

**Comment by JulianOzelRose** [comment]
According to [this thread](https://www.tombraiderforums.com/showthread.php?t=225252), the shotgun in TR2 does 3 damage, with a x1 or x6 multiplier that is based on distance. I tried to find the base shotgun ammo damage value, and it does not seem like it is stored anywhere in the `savegame.dat` file. The global variables are likely hardcoded in the game, or stored in another file. I will take another look later, but it doesn't look like the global variables can be changed from just the savegame file.

Editing entity health/data is 100% possible because that data is definitely stored in the savegame file. What I was saying is that editing "important" entities that only appear once and don't respawn (i.e. final bosses) would be relatively easy to implement. The entities that spawn based on triggers or respawn after death are tricky because their entity data shifts around a lot. Also, there is somewhere around 30 unique entities just in TR2, so it would take a long time to find all of that data. Since there are only 4 final bosses in TR2, I think I could write a build pretty quickly for that though.

In short, getting it to edit final boss entity data would be relatively easy. Getting it to edit all entity data would take a very long time.

**Comment by Menghus** [comment]
Sure, you can take a look to find out more about where the global variables are stored exactly, but I don't want this to turn out to be too much of a hassle and time consuming task for you. If that turns out to be the case then it's totally fine if you abandon this task and leave everything as it is with the savegame editor. This entire idea was meant to be just something extra to improve the combat in the three games but it's not super important and I personally would be totally fine without this.

You can investigate for yourself and estimate if this is worth it or not. To be honest, I don't have much use to edit boss entity health while everything else possibly stays the same. If that's the case then don't bother. If you discover some more information that I need to know about, please post a comment here.

**Comment by JulianOzelRose** [comment]
I'm going to have a lot less free time in the coming weeks/months. So I won't be able to add complex new features, unfortunately.

However, I am still willing to maintain the repo and tend to bug reports, crashes, or small feature requests (like adding a 'Max Stats' button or something like that). So that being said, if you come across any bugs/errors/crashes, feel free to open up another issue and I will take care of it :)
